File: sdks/dotnet/sulfone-helium/Domain/Template/Input.cs
using sulfone_helium.Domain.Core;

namespace sulfone_helium.Domain.Template;

public record TemplateInput(
    Dictionary<string, IAnswer> Answers,
    Dictionary<string, string> DeterministicState
);

public record TemplateValidateInput(
    Dictionary<string, IAnswer> Answers,
    Dictionary<string, string> DeterministicState,
    string Validate
);

File: sdks/node/src/domain/template/input.ts
import type { Answer } from '../core/answer.js';

interface TemplateInput {
  readonly answers: Record<string, Answer>;
  readonly deterministicState: Record<string, string>;
}

interface TemplateValidateInput {
  readonly answers: Record<string, Answer>;
  readonly deterministicState: Record<string, string>;
  readonly validate: string;
}

// export all
export type { TemplateInput, TemplateValidateInput };
